coral-visualization/src/test/java/com/linkedin/coral/vis/VisualizationUtilTest.java

```diff
@@ -18,27 +26,103 @@
     HiveMscAdapter mscAdapter = createMscAdapter();
     HiveToRelConverter converter = new HiveToRelConverter(mscAdapter);
     VisualizationUtil visualizationUtil = VisualizationUtil.create(imagesTempDir);
-    visualizationUtil.visualizeCoralSqlNodeToFile(converter.toSqlNode("SELECT * FROM foo, bar WHERE a = 1"),
-        "/test1.png");
-    // For now we test the presence of the file.
-    // In the future, we can add tests to validate the structure of the generated graph by inspecting the corresponding
-    // JSON file.
-    assertEquals(imagesTempDir.list().length, 1);
+    for (int i = 0; i < queries.length; i++) {
+      visualizationUtil.visualizeCoralSqlNodeToFile(converter.toSqlNode(queries[i]), "/test" + i + ".png");
+    }
+    assertEquals(imagesTempDir.list().length, 2);
     imagesTempDir.delete();
   }
 
+  @Test
+  public void testBasicQueryJson() {
+    JsonObject jsonObject = getJsonObject("SELECT * FROM foo, bar WHERE a = 1");
+    assertTrue(jsonLabelsExist(jsonObject, "SELECT", "JOIN", "=", "LIST", "foo", "bar", "a", "1", "*"));
+  }
+
+  @Test
+  public void testLateralJoinQueryJson() {
+    JsonObject jsonObject = getJsonObject(
+        "SELECT key, value FROM (SELECT MAP('key1', 'value1') as m) tmp LATERAL VIEW EXPLODE(m) m_alias AS key, value");
+    assertTrue(jsonLabelsExist(jsonObject, "SELECT", "JOIN", "LATERAL", "UNNEST", "AS", "MAP", "LIST"));
+  }
+
+  private JsonObject getJsonObject(String s) {
+    HiveMscAdapter mscAdapter = createMscAdapter();
+    HiveToRelConverter converter = new HiveToRelConverter(mscAdapter);
+    String jsonString = convertToGraphVisJsonString(converter.toSqlNode(s));
+    Gson gson = new Gson();
+    return gson.fromJson(jsonString, JsonObject.class);
+  }
+
+  private static String convertToGraphVisJsonString(SqlNode sqlNode) {
+    SqlVisitor<Node> sqlVisitor = new SqlNodeVisualizationVisitor();
+    Node node = sqlNode.accept(sqlVisitor);
+    Graph graph =
+        Factory.graph("test").directed().nodeAttr().with(Font.size(10)).linkAttr().with(Font.size(10)).with(node);
+    return Graphviz.fromGraph(graph).width(1000).render(Format.JSON).toString();
+  }
+
+  private static boolean jsonLabelsExist(Object jsonObject, String... labels) {
+    String[] keys = new String[labels.length];
+    Arrays.fill(keys, "label");
+    return jsonKeyValuePairsExist(jsonObject, keys, labels);
+  }
+
+  private static boolean jsonKeyValuePairsExist(Object object, String[] keys, String[] values) {
+    for (int i = 0; i < keys.length; i++) {
+      if (!jsonKeyValueExists(object, keys[i], values[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private static boolean jsonKeyValueExists(Object object, String key, String value) {
+    if (object instanceof JsonObject) {
+      JsonObject jsonObject = (JsonObject) object;
+      if (jsonObject.has(key)) {
+        if (jsonObject.get(key).getAsString().equals(value)) {
+          return true;
+        }
```

Comment:
nit: can be simplified to be `return jsonObject.get(key).getAsString().equals(value);`